pkg/bgpv1/manager/reconciler/advertisements.go

```diff
@@ -1,14 +1,15 @@
 // SPDX-License-Identifier: Apache-2.0
 // Copyright Authors of Cilium
 
-package manager
+package reconciler
 
 import (
 	"context"
 	"fmt"
 
 	"github.com/sirupsen/logrus"
 
+	"github.com/cilium/cilium/pkg/bgpv1/manager"
```

Comment:
Is it possible for the reconciler to not import manager? It seems incorrect for the inner package to import higher-level package. 

I support this would mean moving various other components of manager to individual packages.

Reply:
Good point, I have tried that in this commit: https://github.com/cilium/cilium/pull/29277/commits/75ac0d931e633f84d65a4a2008cfedb7e9e8a4ce

This however is still not ideal - I had to move [NewServerWithConfig](https://github.com/cilium/cilium/blob/75ac0d931e633f84d65a4a2008cfedb7e9e8a4ce/pkg/bgpv1/manager/reconciler/instance.go#L21-L32) into the reconciler package, which is not that nice, as it is being called from the `manager` as well as the `reconciler` (`PreflightReconciler`) packages now.. and this feels a bit weird to me.

What we could do about it:
1. separate the actual GoBGP server creation and ServerWithConfig construction, so that the `manager` (and `PreflightReconciler`) calls:
```go
s, err := gobgp.NewGoBGPServerWithConfig(ctx, log, params)
sCfg := reconciler.ServerWithConfig(s)
``` 
This does not change the fact that the GoBGP server is created from both packages, but makes it a bit more explicit and cleaner?

2. (in an addition to 1) Instead of recreating the GoBGP server from the `PreflightReconciler`, `Reconcile()` could have a new return value telling the `manager` whether the server needs to be recreated? That way the GoBGP server would be always (re)created only from the `manager`

3. we keep it like it is in this attempt

4. we do not split the reconcilers from the manager?

Reply:
> 1. separate the actual GoBGP server creation and ServerWithConfig construction, so that the `manager` (and `PreflightReconciler`) calls:
> 
> ```go
> s, err := gobgp.NewGoBGPServerWithConfig(ctx, log, params)
> sCfg := reconciler.ServerWithConfig(s)
> ```
> 
I don't mind this approach, it is explicit - we can create new instance package under /manager/instance
> 2. (in an addition to 1) Instead of recreating the GoBGP server from the `PreflightReconciler`, `Reconcile()` could have a new return value telling the `manager` whether the server needs to be recreated? That way the GoBGP server would be always (re)created only from the `manager`

I'd avoid this approach in this PR, since it will introduce logic change - it probably would warrant dedicated change.

> 3. we keep it like it is in this attempt

I am not opposed to this idea - I'm thinking of refactoring like (1) for bgpv2. 

> 4. we do not split the reconcilers from the manager?

Totally upto you, I indent to do it anyways with bgpv2. This refactoring helps me in the sense that change is not going to be too drastic with bgpv2.